framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -4651,21 +4795,36 @@ public boolean isValidUse(AnnotatedArrayType type, Tree tree) {
      * check the return type.
      *
      * @param tree the AST type supplied by the user
+     * @return true if the type is valid
      */
     public boolean validateTypeOf(Tree tree) {
         AnnotatedTypeMirror type;
         // It's quite annoying that there is no TypeTree.
         switch (tree.getKind()) {
             case PRIMITIVE_TYPE:
             case PARAMETERIZED_TYPE:
-            case TYPE_PARAMETER:
             case ARRAY_TYPE:
             case UNBOUNDED_WILDCARD:
             case EXTENDS_WILDCARD:
             case SUPER_WILDCARD:
             case ANNOTATED_TYPE:
                 type = atypeFactory.getAnnotatedTypeFromTypeTree(tree);
                 break;
+            case TYPE_PARAMETER:
+                type = atypeFactory.getAnnotatedTypeFromTypeTree(tree);
+                for (Tree t : ((TypeParameterTree) tree).getBounds()) {
+                    validateTargetLocation(
+                            t,
+                            ((AnnotatedTypeVariable) type).getUpperBound(),
+                            TypeUseLocation.UPPER_BOUND,
+                            TypeUseLocation.EXPLICIT_UPPER_BOUND);
+                }
+                validateTargetLocation(
+                        tree,
+                        ((AnnotatedTypeVariable) type).getLowerBound(),
+                        TypeUseLocation.LOWER_BOUND,
+                        TypeUseLocation.EXPLICIT_LOWER_BOUND);
```

Comment:
remove `EXPLICIT_LOWER_BOUND`. make this method accept only 1 type-use location parameter